webapp/src/js/enketo/medic-xpath-extensions.js

```diff
@@ -14,11 +14,54 @@ var getValue = function(resultObject) {
   return resultObject.v;
 };
 
+var now_and_today = function() { return { t: 'date', v: new Date() }; };
```

Comment:
I wonder if these should be two different types: one for `date` and one for `datetime`. That way you wouldn't have to assume if the datetime ends with `T00:00:00.000` then it's a date.